test/dns-test-srv/main.go

```diff
@@ -149,7 +149,7 @@ func (ts *testSrv) serveTestResolver() {
 	dns.HandleFunc(".", ts.dnsHandler)
 	dnsServer := &dns.Server{
 		Addr:         "0.0.0.0:8053",
-		Net:          "tcp",
+		Net:          "udp",
```

Comment:
I think this might also need to be gated by the feature flag so that tests will work against the non-next config.

Reply:
Or maybe its easier to not pipe all the feature flag stuff into this test server and just listen on both udp/tcp?